### PR TITLE
Add new material to service worker section.

### PR DESCRIPTION
**Pull request by jpmedley**
Partially addresses: https://github.com/GoogleChromeLabs/Extension-Docs/issues/15

### CODE REVIEW

File: site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md [hunk 1]
```diff
@@ -41,17 +41,15 @@ When a user profile starts, the [`chrome.runtime.onStartup`](/docs/extensions/re
...
+Events and calls to extension APIs reset these timers, and if the service worker has gone dormant, they will revive them. Nevertheless, you should design your service worker to be resilient against unexpected termination.
 
-Though Chrome 110 changed this you should not keep your service worker alive indefinitely, though you can. (We don't consider this a good programming practice.) You should test your extensions to ensure that they're not doing this unintentionally.
-
-Though it's difficult to catch in a code review, we suggest you guard against it as best you can.
-
-For all of these reasons, you need to guard against unexpected termination of the service worker.
+You should not keep your service worker alive indefinitely, though you can. (We don't consider this a good programming practice.) You should test your extensions to ensure that they're not doing this unintentionally. Though it's difficult to catch in a code review, we suggest you guard against it as best you can.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md:52]: Suggestion: I think we can simplify this paragraph to the following:

"To optimize the resource consumption of your extension, avoid keeping your service worker alive indefinitely if possible. Test your extensions to ensure that you're not doing this unintentionally."

File: site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md [hunk 2]
```diff
@@ -66,6 +64,22 @@ Any global variables you set will be lost if the service worker shuts down. Inst
 [CacheStorage API](https://developer.mozilla.org/docs/Web/API/CacheStorage)
 : A persistent storage mechanism for Request and Response object pairs. This API was designed specifically for web service workers and is used to retrieve data from an endpoint. There are a variety of ways to use this API depending on whether and how critical it is that users see up-to-date data. For more information, see [The Offline Cookbook](​​https://web.dev/offline-cookbook). Unless you're specifically proxying network requests via the fetch handler, you should use `chrome.storage`.
 
-## Be careful with timeouts {: #timeouts }
+### Keeping service workers alive
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md:67]: When did Chrome start resetting the timer on extension API calls?

**Review by sebastianbenz — COMMENTED** [review]
Great updates! Left a few comments.

File: site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md (hunk 1 above)
Comment (jpmedley) [reply site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md:52]: Done.

File: site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md:67]: Done.

Comment (jpmedley) [reply site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md:67]: That's in 110. I tweaked the language to make that clearer.

**Comment by jpmedley** [comment]
PTAL

File: site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md [hunk 3]
```diff
@@ -68,6 +68,15 @@ Any global variables you set will be lost if the service worker shuts down. Inst
 
 Since the release of Manifest V3, we've made several improvements to service worker lifetimes. This means that if your Manifest V3 extension supports earlier versions of Chrome, there are conditions you will need to be aware of. If these conditions do not affect your extension, you can move on from this section. If they do, consider specifying a [minimum Chrome version](/docs/extensions/mv3/manifest/minimum_chrome_version/) in your manifest.
 
+#### Chrome 116
+
+Chrome 116 introduced the following service worker lifetime improvements:
+
+* Active [`WebSocket`](https://developer.mozilla.org/docs/Web/API/WebSockets_API) connections no extend extension service worker lifetimes. Sending or receiving messages across a `WebSocket` in an extension service worker resets the service worker's idle timer.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md:75]:
```suggestion
* Active [`WebSocket`](https://developer.mozilla.org/docs/Web/API/WebSockets_API) connections now extend extension service worker lifetimes. Sending or receiving messages across a `WebSocket` in an extension service worker resets the service worker's idle timer.
```

**Review by AmySteam — CHANGES_REQUESTED** [review]
Thanks @jpmedley! 💯  I just a few minor tweaks 👍

File: site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md [hunk 4]
```diff
@@ -66,6 +64,31 @@ Any global variables you set will be lost if the service worker shuts down. Inst
 [CacheStorage API](https://developer.mozilla.org/docs/Web/API/CacheStorage)
 : A persistent storage mechanism for Request and Response object pairs. This API was designed specifically for web service workers and is used to retrieve data from an endpoint. There are a variety of ways to use this API depending on whether and how critical it is that users see up-to-date data. For more information, see [The Offline Cookbook](​​https://web.dev/offline-cookbook). Unless you're specifically proxying network requests via the fetch handler, you should use `chrome.storage`.
 
-## Be careful with timeouts {: #timeouts }
+### Choose a minimum Chrome version
+
+Since the release of Manifest V3, we've made several improvements to service worker lifetimes. This means that if your Manifest V3 extension supports earlier versions of Chrome, there are conditions you will need to be aware of. If these conditions do not affect your extension, you can move on from this section. If they do, consider specifying a [minimum Chrome version](/docs/extensions/mv3/manifest/minimum_chrome_version/) in your manifest.
+
+#### Chrome 116
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md:71]: Sebastian already told me he thought this was OK.

File: site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md [hunk 5]
```diff
@@ -4,7 +4,7 @@ title: The extension service worker lifecycle
 subhead: 
 description: Extension service workers respond to both standard service worker events and events in extension namespaces. They are presented together because often one type follows another during an extension's use.
 date: 2023-05-02
-updated: 2023-05-03
+updated: 2023-06-22
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md:7]: Done.

File: site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md [hunk 6]
```diff
@@ -66,6 +64,31 @@ Any global variables you set will be lost if the service worker shuts down. Inst
...
+Extension API calls reset the timers. Before this, only running event handlers would keep a service worker alive. Any events that were queued, but for which a handler had not been called would not cause a reset.
+
+#### Chrome 109
+
+Messages sent from an offscreen document reset the timers.
+
+#### Chrome 105
 
-If an operation takes more than 30 seconds to complete, the service worker can shut down. An example is the `fetch()` call described above. The fetch fails if the service worker shuts down before the [`response`](https://developer.mozilla.org/docs/Web/API/Response) is received.
+Connecting to a native messaging host using chrome.runtime.connectNative() will keep a service worker alive. If the host process crashes or is shut down, the port is closed and the service worker will terminate after timers complete. Guard against this by calling chrome.runtime.connectNative() in the port's onDisconnect event handler.
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md]: Done.

**Comment by jpmedley** [comment]
@AmySteam PTAL

**Review by AmySteam — APPROVED** [review]
Thanks Joe 💯

**Pull request merged**